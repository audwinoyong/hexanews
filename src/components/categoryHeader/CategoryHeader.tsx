import * as React from 'react'
import { Link } from 'react-router-dom'
import AppBar from '@material-ui/core/AppBar'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'

import styles from './styles'

type Props = {
}

class CategoryHeader extends React.Component<Props> {

  constructor(props: Props) {
    super(props)
  }

  renderCategoriesList = () => {
    const categories = ['tech', 'science', 'art', 'design', 'culture',
      'photography', 'leadership', 'math', 'economy', 'music', 'more']
    return categories.map((item, i) => {
      return (
        <Typography variant='button' color='primary' style={styles.flex}>
          <Link to='/' style={styles.category}>{item.toUpperCase()}</Link>
        </Typography>
      )
    })
  }

  public render() {

    return (
      <div style={styles.root}>
        <AppBar position='static' style={styles.appBar}>
          <Toolbar>
            {this.renderCategoriesList()}
          </Toolbar>
        </AppBar>
      </div>
    )
  }
}

export default CategoryHeader