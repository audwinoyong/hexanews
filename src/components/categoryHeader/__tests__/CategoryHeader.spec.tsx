import * as React from 'react'
import * as renderer from 'react-test-renderer'
import * as Enzyme from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'
import { BrowserRouter } from 'react-router-dom'

import CategoryHeader from '../CategoryHeader'

Enzyme.configure({ adapter: new Adapter() })

describe('Category Header', () => {
  it('renders Category Header', () => {
    const component = renderer.create(
      <BrowserRouter>
        <CategoryHeader
          categories={[]}
        />
      </BrowserRouter>
    )
    let tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})