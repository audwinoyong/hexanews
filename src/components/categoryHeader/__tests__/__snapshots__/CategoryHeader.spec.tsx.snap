// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Category Header renders Category Header 1`] = `
<div
  className=""
  style={
    Object {
      "flexGrow": 1,
      "overflow": "auto",
    }
  }
>
  <header
    className="MuiPaper-root-106 MuiPaper-elevation4-112 MuiAppBar-root-98 MuiAppBar-positionStatic-102 MuiAppBar-colorPrimary-104"
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "boxShadow": "none",
        "paddingRight": 18,
      }
    }
  >
    <div
      className="MuiToolbar-root-133 MuiToolbar-regular-135 MuiToolbar-gutters-134"
    />
  </header>
</div>
`;
